### ISSUE TITLE
Out Of Memory on parsing\generation big Excel

### ISSUE THREAD

**Issue by bubnenkoff** [syncfusion/flutter-widgets#448]
I have got 71MB Excel file with 365 000 of rows. I need extract only one of them.
The problem that xlsio is fail in browser with OOM error.

Same OOM error are happens when I am trying to generate Excel from 80MB JSON file.

**Comment by KonduruKeerthi** [comment]
Greetings from Syncfusion.

This is to inform you that Syncfusion Flutter XlsIO works fine when you try to create an Excel document with 3,65,000 rows and 20 columns. Increasing the column counts beyond 20 columns would result in performance or do throw exception.

You have not mentioned the number of columns you are importing. Kindly let us know the column count at your end.

Also, you have mentioned that you are trying to extract only one of them. We are sorry to inform you that we have not understood this clearly. Are you trying to import the entire JSON data and extract one row from that? Please clarify.

If you are using more than 20 columns for 365,000 rows, Syncfusion Flutter XlsIO would not perform as expected. This is the limit. Hence we request you to reduce the column count.

Regards,
Keerthi.

**Comment by bubnenkoff** [comment]
@KonduruKeerthi I have only 11 columns in my document. I am using follow code to generate file:
```
Future<void> generateExcel(List data) async {
  //Creating a workbook.
  final Workbook workbook = Workbook();
  //Accessing via index
  final Worksheet sheet = workbook.worksheets[0];
  Style globalStyle = workbook.styles.add('style');
  globalStyle.wrapText = true;
  globalStyle.hAlign = HAlignType.center;

  Range range = sheet.getRangeByName('A1:K1');
  range.cellStyle = globalStyle;  

  sheet.getRangeByName('A1').columnWidth = 20;
  sheet.getRangeByName('B1').columnWidth = 20;
  sheet.getRangeByName('C1').columnWidth = 12;
  sheet.getRangeByName('D1').columnWidth = 12;
  sheet.getRangeByName('E1').columnWidth = 16;
  sheet.getRangeByName('F1').columnWidth = 16;
  sheet.getRangeByName('G1').columnWidth = 15;
  sheet.getRangeByName('H1').columnWidth = 18;
  sheet.getRangeByName('I1').columnWidth = 12;
  sheet.getRangeByName('J1').columnWidth = 32;
  sheet.getRangeByName('K1').columnWidth = 12;

  sheet.showGridlines = false;

  // Enable calculation for worksheet.
  sheet.enableSheetCalculations();

  sheet.showGridlines = false;
  
  sheet.getRangeByName('A1:K1').cellStyle.backColor = '#edf5ff';
  sheet.getRangeByName('A1:K1').cellStyle.bold = true;
  sheet.getRangeByName('A1').setText('Номер Игрока');
  sheet.getRangeByName('B1').setText('Извещение');
  sheet.getRangeByName('C1').setText('Количество ');
  sheet.getRangeByName('D1').setText('Количество новых');
  sheet.getRangeByName('E1').setText('Дата первой версии ');
  sheet.getRangeByName('F1').setText('Дата внесения изменений');
  sheet.getRangeByName('G1').setText('Сколько подано заявок');
  sheet.getRangeByName('H1').setText('Сколько допущено ');
  sheet.getRangeByName('I1').setText('Игрок');
  sheet.getRangeByName('J1').setText('Включено перечень');
  sheet.getRangeByName('K1').setText('Ставка');


  int rowNumber = 2;
  for(var row in data) {
    print(row.isItemForAuctionInElectronicForm);
    print(row.isWinnerDishonest);
    print('-----');
    sheet.getRangeByName('A$rowNumber').setText(row.contractRegistryNumber);
    sheet.getRangeByName('B$rowNumber').setText(row.purchaseNumber);
    sheet.getRangeByName('C$rowNumber').setNumber(row.complaintsCount);
    sheet.getRangeByName('D$rowNumber').setNumber(row.issuedWarrantsCount);
    sheet.getRangeByName('E$rowNumber').setDateTime(row.firstNotificationPlacementDate);
    sheet.getRangeByName('F$rowNumber').setDateTime(row.lastNotificationModificationDate);
    sheet.getRangeByName('G$rowNumber').setNumber(row.participantsApplcationsCount);
    sheet.getRangeByName('H$rowNumber').setNumber(row.applcationsAcceptedCount);
    sheet.getRangeByName('I$rowNumber').setText(row.isWinnerDishonest.toString().replaceFirst('false', 'нет').replaceFirst('true', 'да'));
    sheet.getRangeByName('J$rowNumber').setValue(row.isItemForAuctionInElectronicForm.toString().replaceFirst('false', 'нет').replaceFirst('true', 'да'));
    sheet.getRangeByName('K$rowNumber').setText(row.advance);

    rowNumber++;
  }


    //Save and launch the excel.
    final List<int> bytes = workbook.saveAsStream();
    //Dispose the document.
    workbook.dispose();

    //Save and launch the file.
    await saveAndLaunchFile(bytes, 'nprojects.xlsx');
  }
```
[data.zip](https://github.com/syncfusion/flutter-widgets/files/7518674/data.zip)

**Comment by KonduruKeerthi** [comment]
We appreciate your patience.

We are unable to reproduce the issue even with the given code snippet. We have checked the issue in the Web application and found that it takes around 2.5GB of memory approximately for Excel generation. Kindly confirm us whether you are using the web application or a mobile application. This will be helpful for us in investigating the query.

Regards,
Keerthi.

**Comment by bubnenkoff** [comment]
I use flutter for web on desktop.
Could you share your test project? I will try to build and run it

Пт, 31 дек. 2021 г. в 14:09, KonduruKeerthi ***@***.***>:

**Comment by KonduruKeerthi** [comment]
Hi @bubnenkoff ,

We appreciate your patience.

This is to inform you that we have included a fix for some memory-related issues in our latest weekly NuGet release version **19.4.42**. The sample which we have tried at our end can be downloaded from the below link.

**Sample Link:** https://www.syncfusion.com/downloads/support/directtrac/general/ze/flutter_memoryoutofexception_1-2125890765.zip 

We request you to check the issue in our latest version and let us know the results.

Regards,
Keerthi.
